### ISSUE TITLE
about function sample()

### ISSUE THREAD

**Issue by liiiiiiiiil** [ruotianluo/self-critical.pytorch#45]
hello,RuoTianLuo.I am confused about the function sample().could you please tell me what this function do, Monte Carlo?

**Comment by lukemelas** [comment]
The `sample` function in each model picks each word by sampling from distribution over the vocab produced by the softmax (applied to the output of the LSTM). If `sample_max` is true, then the model does greedy search (instead of sampling, it takes the word with the maximum probability and feeds that word in at the next time step). The sample function does not do Monte Carlo estimation.

**Comment by liiiiiiiiil** [comment]
@lukemelas thank you for your answer!
